src/Engine/Physics/Shape.hpp

```diff
@@ -42,8 +48,31 @@ namespace Physics {
              */
             btCollisionShape* GetShape();
 
+            /// Get the type of wrapped shape.
+            /**
+             * @return The type of shape.
+             */
+            Kind GetKind() const;
+
+            /// Get sphere data of the shape.
+            /**
+             * @return Sphere data, or nullptr if the shape is not a sphere.
+             */
+            const Sphere* GetSphereData() const;
+
+            /// Get plane data of the shape.
+            /**
+             * @return Plane data, or nullptr if the shape is not a plane.
+             */
+            const Plane* GetPlaneData() const;
+
         private:
             btCollisionShape* shape;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Class 'Shape' is unsafe, 'Shape::shape' can leak by wrong usage. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunsafeClassCanLeak)